File: src/teste-digi-v2/app/Repository/ProductRepository.php
<?php

namespace App\Repository;

use App\Models\Product;
use Illuminate\Support\Facades\File;

class ProductRepository
{
    private $product;

    public function __construct()
    {
        $content = File::json(base_path('app\Data\products.json'));
        $this->product = Product::hydrate($content);
    }

    public function getProducts() {
        return $this->product->all();
    }
}

File: src/TesteDigi/TesteDigi.Repository/ProductRepository.cs
using TesteDigi.Repository.Data;
using TesteDigi.Repository.Model;

namespace TesteDigi.Repository
{
    public class ProductRepository
    {
        private ProductData data;

        public ProductRepository() 
        {
            data = new ProductData();
        }

        public List<Product> GetAll()
        {
            return data.GetProducts();
        }
    }
}
